14 TestComponent/src/components/member/spec/memberPage.spec.tsx

```diff
@@ -0,0 +1,444 @@
+            saveMember: saveMemberMock
+        };
+
+        let memberPageWrapper = shallow(
+            <MemberPage {...properties} />
+        );
+
+        memberPageWrapper.simulate('save', {
+            preventDefault: eventPreventDefaultMock
+        })
+
+        expect(saveMemberMock.called).to.be.true;
+        expect(saveMemberMock.calledWith(member)).to.be.true;
+        expect(eventPreventDefaultMock.called).to.be.true;
+    });
+
+    it('should does not call to componentWillReceiveProps' +
+        'passing required properties with default values', sinon.test(() => {
+        let sinon: Sinon.SinonStatic = this;
+        let initializeNewMemberMock = sinon.spy();
+        let componentWillReceivePropsMock = sinon.stub(MemberPage.prototype, 'componentWillReceiveProps');
+
+        let properties = {
+            initializeNewMember: initializeNewMemberMock
+        };
+
+        let memberPageWrapper = shallow(
+            <MemberPage {...properties} />
+        );
+
+        expect(componentWillReceivePropsMock.calledOnce).to.be.false;
+    }).bind(this));
+
+    it('should does not call to componentWillReceiveProps' +
+        'passing saveCompleted equals false', sinon.test(() => {
+        let sinon: Sinon.SinonStatic = this;
+        let initializeNewMemberMock = sinon.spy();
+        let componentWillReceivePropsMock = sinon.stub(MemberPage.prototype, 'componentWillReceiveProps');
+
+        let properties = {
+            saveCompleted: false,
+            initializeNewMember: initializeNewMemberMock
+        };
+
+        let memberPageWrapper = shallow(
+            <MemberPage {...properties} />
+        );
+
+        expect(componentWillReceivePropsMock.calledOnce).to.be.false;
+    }).bind(this));
+
+    it('should calls to componentWillReceiveProps but does not calls to toastr.success()' +
+        'passing saveCompleted equals false and setting root component props with ' +
+        'saveCompleted property equals false', sinon.test(() => {
+        let sinon: Sinon.SinonStatic = this;
+        let initializeNewMemberMock = sinon.spy();
+        let componentWillReceivePropsMock = sinon.stub(MemberPage.prototype, 'componentWillReceiveProps');
+        let toastrMock = sinon.stub(toastr, 'success');
+
+        let properties = {
+            saveCompleted: false,
+            initializeNewMember: initializeNewMemberMock
+        };
+
+        let memberPageWrapper = shallow(
+            <MemberPage {...properties} />
+        );
+
+        memberPageWrapper.setProps({
+            saveCompleted: false
+        });
+
+        expect(componentWillReceivePropsMock.calledOnce).to.be.true;
+        expect(toastrMock.calledOnce).to.be.false;
+    }).bind(this));
+
+    it('should calls to componentWillReceiveProps but does not calls to toastr.success()' +
+        'passing saveCompleted equals true and setting root component props with ' +
+        'saveCompleted property equals true', sinon.test(() => {
```

Comment:
Add a comment "sinson.test" make automatic cleanup and add a link explaining how it works, we will do this just in one place to help people know about this